packages/page-builder/src/containers/CollectionPage/index.tsx

```diff
@@ -59,6 +64,13 @@ function CollectionPage ({ account, basePath }: CollectionPageProps): ReactEleme
     }
   }, [collectionId, getDetailedCollectionInfo]);
 
+  // set previewButtonDisplayed to AppContext
+  useEffect(() => {
+    setPreviewButtonDisplayed(previewMode);
```

Comment:
Please do not call setState from the useEffect hook. Use callback for it. Also use IsMountedRef check

Reply:
почему?

Reply:
Seems correct. We are subscribing to threshold changes. When they happen - update displayability

Not sure why not just use previewMode instead of putting it into state, though